src/main/kotlin/stringCalculator/StringAddCalculator.kt

```diff
@@ -8,18 +8,25 @@ class StringAddCalculator {
 
         val matchedCustomDelimiter = CUSTOM_DELIMITER_REGEX.find(inputValue)
 
-        val stringList = matchedCustomDelimiter?.let { split(it.groupValues[1], it.groupValues[2]) } ?: split(inputValue)
+        val numberStrings = extractNumberStrings(inputValue, matchedCustomDelimiter)
 
-        validateNumber(stringList)
+        validateNumber(numberStrings)
 
-        return sum(stringList)
+        return sum(numberStrings)
     }
 
-    private fun split(string: String): List<String> {
-        return string.split(DEFAULT_DELIMITER_REGEX)
+    private fun extractNumberStrings(inputValue: String, matchedCustomDelimiter: MatchResult?): List<String> {
+        if (matchedCustomDelimiter == null) {
+            return split(inputValue)
+        }
+        val string = matchedCustomDelimiter.groupValues[2]
+        val delimiter = matchedCustomDelimiter.groupValues[1]
```

Comment:
string이라는 변수명보다는, 다른 변수명이 좋을거같아요! 

destructured를 활용해보셔도 좋을거같아요 👍 
```suggestion
        val (string: String, delimiter: String) = matchedCustomDelimiter.destructured
```

Reply:
오 신기하네요